java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -4012,20 +4012,29 @@ static ColumnVector createColumnVector(DType type, int rows, HostMemoryBuffer da
       }
       if (mainColOffsets != null) {
         // The offset buffer has (no. of rows + 1) entries, where each entry is INT32.sizeInBytes
-        long offsetsLen = OFFSET_SIZE * (mainColRows + 1);
+        long offsetsLen = ((long)OFFSET_SIZE) * (mainColRows + 1);
```

Comment:
We should upcast `mainColRows` (or just have it declared as a `long`) rather than upcast `OFFSET_SIZE` to cover the case where `mainColRows` == `Integer.MAX_VALUE`, overflowing before the implicit upcast by the multiply.

Reply:
Technically the number of mainColRows cannot be `Integer.MAX_VALUE` or CUDF would overflow when trying to create the offsets column. This was here to just clean up a warning.  But I will fix it because it is cleaner/more correct.